internal/compiler/multierror.go

```diff
@@ -0,0 +1,34 @@
+package compiler
+
+import "strings"
+
+// MultiError is a wrapper for an error slice, which provides convenient
+// wrapping of multiple errors into a single error. MultiError is not safe for
+// concurrent use.
+type MultiError struct {
+	errs []error
+}
+
+// Append appends an error to the multi error.
+func (e *MultiError) Append(err error) {
+	if err == nil {
```

Comment:
Will we ever append without checking the error?

Reply:
Probably not.